content/chainguard/chainguard-images/how-to-use/dev-containers/index.md

```diff
@@ -0,0 +1,223 @@
+
+Assuming you're starting from a project with no dev container, you'll first need to create the requisite files and folders.
+
+Start by creating a directory named `.devcontainer` and navigate into it:
+
+```shell
+mkdir .devcontainer && cd $_
+```
+
+Next, create a dev container configuration file named `devcontainer.json` with the following command:
+
+```shell
+cat > devcontainer.json <<EOF
+{
+	"name": "my-devcontainer",
+	"build": {
+    	"dockerfile": "Dockerfile",
+    	"args": {}
+	},
+	"customizations": {
+    	"vscode": {
+        	"extensions": [ "ms-python.python" ]
+    	}
+	},
+	"postCreateCommand": "pip install -r requirements.txt",
+	"remoteUser": "nonroot"
+}
+
+EOF
+```
+
+Because this is for a Python app, we are installing the VS Code `python` extension and using `pip` to install project dependencies.
+
+The `postCreateCommand` runs in the root of your project after it has been cloned into the container. If you don't have a `requirements.txt` file with your project the command will fail and you will need to remove the command to use the dev container.
+
+Following that, you'll need to find a base image. Chainguard offers a wide range of images for different languages and ecosystems. To search, use [the images directory](https://images.chainguard.dev/directory?category=languages). For this example, Chainguard's Python image will suffice.
+
+Create a Dockerfile with the following command:
+
+```shell
+cat > Dockerfile <<EOF
+FROM chainguard/python:latest-dev
+
+USER root
+RUN apk update && apk add posix-libc-utils && ldconfig
+USER nonroot
+RUN pip install pylance debugpy
+
+EOF
+```
+
+Here's what each line of this Dockerfile does:
+
+* `FROM`: Many Chainguard images are available at Docker Hub, meaning you can set the `FROM` line in the `Dockerfile` like this example. Alternatively, you could set this with `cgr.dev/chainguard/python` or, if your Chainguard organization has access to a Production Python image, `cgr.dev/$ORGANIZATION/python:$TAG`.
+* `USER root`: Including this line forces the next commands to run as root instead of the unprivileged user. Whether you need to include this line or not will depend on what image you're using.
+    * You can check by running your image locally and using the `id` command. The following example shows that the Python image uses the nonroot user by default while the Go image uses root by default:
+
+``` shell
+docker run -it --entrypoint id chainguard/python:latest-dev
+```
+```Output
+uid=65532(nonroot) gid=65532(nonroot) groups=65532(nonroot)
+docker run -it --entrypoint id chainguard/go:latest-dev
+uid=0(root) gid=0(root) groups=0(root),1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)
+```
```

Comment:
Do we indent example code like this, that belongs to the bullet point above? I couldn't see anything about this in the style guide.

Reply:
I didn't think indenting code blocks was possible, but apparently it is!  Thanks for flagging, it looks much much better now.

I don't think there's anything specific about this in the style guide. Maybe worth adding something though.